src/Sentry/MetricAggregator.cs

```diff
@@ -334,9 +370,27 @@ public async Task FlushAsync(bool force = true, CancellationToken cancellationTo
                 cancellationToken.ThrowIfCancellationRequested();
 
                 _options.LogDebug("Flushing metrics for bucket {0}", key);
-                if (!Buckets.TryRemove(key, out var bucket))
+                ConcurrentDictionary<string, Metric>? bucket;
+                _bucketsLock.EnterWriteLock();
+                try
+                {
+#if NETSTANDARD2_1_OR_GREATER
+                    if (!Buckets.Remove(key, out bucket))
+                    {
+                        continue;
+                    }
+#else
+                    if (!Buckets.ContainsKey(key))
+                    {
+                        continue;
+                    }
+                    bucket = Buckets[key];
+                    Buckets.Remove(key);
+#endif
```

Comment:
nit? Maybe just keep a single variant of the code? Should work in any .net version, I think?
```suggestion
                    if (!Buckets.ContainsKey(key))
                    {
                        continue;
                    }
                    bucket = Buckets[key];
                    Buckets.Remove(key);
```